### PR TITLE
Add payment service integration

### PR DESCRIPTION
**Pull request by hepsimo**
Add support for user web journey through GOV.pay

- add payment template collection `payment_templates` to mongodb; include JSON data file
- refactor payment report generation -- removing 'successful submissions report' to Finance, and using last sessionId in payment_sessions BI-5190
- add `/payment` endpoint called by payment service
- add `/payment-sessions` endpoint

### Database Note:
You’ll need to do apply the following changes to the efs_submissions database manually:
- create the mongo collection `payment_templates`
- load data from file `src/main/resources/payment_templates.json`
- reload data from file `src/main/resources/form_templates.json`

### Note:
Requires update to private-api-sdk-java version from https://github.com/companieshouse/private-api-sdk-java/pull/83. POM dependency version is `unversioned` until that PR is merged and the new version is known.

Requires:
- https://github.com/companieshouse/efs-submission-web/pull/7
- https://github.com/companieshouse/private-api-sdk-java/pull/83

### CODE REVIEW

File: spec/swagger.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "info": {
```
Comment (PhilipGlover) [inline spec/swagger.json:3]: when I run this in dapperbox I see the generateFinanceReports under the payment service section

Screenshot 2020-10-15 at 11 47 32

File: src/main/java/uk/gov/companieshouse/efs/api/config/Config.java [hunk 2]
```diff
@@ -60,8 +59,6 @@
     private String barcodeGeneratorServiceUrl;
     @Value("${file.bucket.name}")
     private String fileBucketName;
-    @Value("${payment.report.bucket.name}")
-    private String paymentReportBucketName;
```
Comment (PhilipGlover) [inline src/main/java/uk/gov/companieshouse/efs/api/config/Config.java:64]: do we need to leave this config in now?
and the beans below

Comment (PhilipGlover) [reply src/main/java/uk/gov/companieshouse/efs/api/config/Config.java]: this has commit has now been reverted, so this is no longer relevant.

**Comment by hepsimo** [comment]
Note: Updated the POM with appropriate SDK release version

File: spec/swagger.json (hunk 1 above)
Comment (PhilipGlover) [reply spec/swagger.json:3]: not a biggy

**Review by PhilipGlover — APPROVED** [review]
run successfully for forms with and without a fee, and when cancel payment and then completing payment.

**Pull request merged**